src/nvim/ex_cmds.c

```diff
@@ -5289,9 +5288,13 @@ void ex_helptags(exarg_T *eap)
   }
 
   /* Get a list of all files in the help directory and in subdirectories. */
-  STRCPY(NameBuff, dirname);
+  STRLCPY(NameBuff, dirname, MAXPATHL);
   add_pathsep((char *)NameBuff);
```

Comment:
This function could use `strcat()` to add to `NameBuff` and overflow the array, right in the middle of all these "safe" functions.